### PR TITLE
Cypress e2e

### LINKED ISSUE #71: add UI e2e testing
#### Description

we can make use of cypress to do the integration testing and have github automate the tests (example: https://github.com/CSCfi/metadata-submitter-frontend/blob/master/.github/workflows/e2etests.yml )

#### DoD (Definition of Done)

e2e tests implemented and added to github actions

#### Testing

Peer review

### PR DESCRIPTION
**Pull request by ViMuilu**
### Description

E2E-test with cypress.

Note you can run the tests with rahti aggregator(https://dev-aggregator-beacon.rahtiapp.fi/) and registry(https://dev-registry-beacon.rahtiapp.fi/) but three of the test will fail. To run all test successfully either run the backend in docker or manually and also populate the db registry with https://staging-elixirbeacon.rahtiapp.fi/ and the aggregator.

I did not add github actions for the test because the back end has some things hardcoded which needs to be changed.

### Related issues

Fixes #71

### Type of change

- [x] New feature (non-breaking change which adds functionality)

### Changes Made

Added cypress test and also new test ids and fixed a small bug where elixir logo was not centred.

### CODE REVIEW

**Comment by ViMuilu** [comment]
Tests fail because npm test -- --coverage clears lcov.info for some reason

**Comment by blankdots** [comment]
this does not fix #71 it provides a testsfor it (solving just part of it)
I think we should focus on developing tests that don't depend on a dev instance but something we can reproduce on an isolated environment.

>  has some things hardcoded which needs to be changed.

can we get a list so that we can fix them, or this is being worked on already?

We can also add more tests:
- test a query that gives no results
- test register form for a beacon.

File: .github/workflows/E2Etests.yml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+          -H 'accept: application/json' \
+          -H 'Content-Type: application/json' \
+          -d '{
+          "type": "beacon-aggregator",
+          "url": "http://app_aggregator:5050/service-info"
+          }'
+      - name: Run Frontend
+        run: |
+          npm install
+          export VUE_APP_AGGREGATOR_URL=http://0.0.0.0:5050/
```
Comment (blankdots) [inline .github/workflows/E2Etests.yml:49]: here you can use `env` https://docs.github.com/en/actions/reference/environment-variables

File: .github/workflows/E2Etests.yml [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+          export VUE_APP_REGISTRY_URL=http://0.0.0.0:8080/
+          npm run build
```
Comment (blankdots) [inline .github/workflows/E2Etests.yml:51]: I know i recommended the python command, but wondering  if `npm serve &` (with the `&` at the end) would have provided similar functionality

**Review by blankdots — COMMENTED** [review]
Sry if i reviewed and was not supposed to, was marked as ready. 
Would be nice to get the commit that were a `fix` and `Update E2Etests.yml ` into 1 commit meaning a force push with more meaningful commits

File: .github/workflows/E2Etests.yml (hunk 2 above)
Comment (ViMuilu) [reply .github/workflows/E2Etests.yml:52]: I think it's better to try to build the project and see if it builds than run it in the dev server.

Comment (blankdots) [reply .github/workflows/E2Etests.yml:52]: fair point

File: .github/workflows/E2Etests.yml (hunk 1 above)
Comment (ViMuilu) [reply .github/workflows/E2Etests.yml]: Done

File: .github/workflows/E2Etests.yml [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node: ["14"]
+        browser: ["firefox", "chrome"]
+    steps:
+      - uses: actions/checkout@v2
+      - name: Setup and Run backend
+        run: |
+          git clone --branch Added_test_docker-compose https://github.com/CSCfi/beacon-network
```
Comment (blankdots) [inline .github/workflows/E2Etests.yml:16]: i think it would be good if we add this branches to at least `dev` branch in `beacon-network` repo.

Comment (ViMuilu) [reply .github/workflows/E2Etests.yml:16]: Yes i'm currently updating that branch

File: .github/workflows/E2Etests.yml [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+      - name: Add beacon to registry
+        run: |
+          curl -X 'POST' \
+            'localhost:8080/services' \
+          -H 'Authorization: 07b4e8ed58a6f97897b03843474c8cc981d154ffe45b10ef88a9f127b15c5c56' \
+          -H 'accept: application/json' \
+          -H 'Content-Type: application/json' \
+          -d '{
+          "type": "beacon",
+          "url": "https://staging-elixirbeacon.rahtiapp.fi/service-info"
```
Comment (blankdots) [inline .github/workflows/E2Etests.yml:33]: did not notice this before - this should not be a real url, or better yet why do we have that real url there?

Comment (ViMuilu) [reply .github/workflows/E2Etests.yml:33]: I think i mentioned this to you before, but the idea was to use the dev beacon as a placeholder and add both the beacon and beaconV2 mocks at the same time when the beaconV2 is implemented. But i can make the mock now if you want it.

Comment (blankdots) [reply .github/workflows/E2Etests.yml:33]: ok, then we can leave it for now and treat the mocks as part of another PR.

**Review by blankdots — APPROVED** [review]

**Comment by ViMuilu** [comment]
The search functionality is broken in the beacon network dev branch. The old "Added test docker compose" branch works fine on my computer but when i change to the dev, search doesn't work anymore. This might be because of beacon network PR#83.

File: cypress/integration/HomeTab.test.ts [hunk 5]
```diff
@@ -0,0 +1,65 @@
+describe("Home tab", () => {
+  describe("Basic search", () => {
+    it("successfully loads", () => {
+      cy.visit("/");
+      cy.contains("Advanced Search");
+    });
+    it("can do a exmaple search", () => {
```
Comment (blankdots) [inline cypress/integration/HomeTab.test.ts:7]:
```suggestion
    it("can do a example search", () => {
```

**Comment by blankdots** [comment]
adding a fix for https://github.com/CSCfi/beacon-network/issues/78 would break these tests - not sure what exactly is the cause, but let us proceed with these tests.

**Pull request merged**